src/webgl/loading.js

```diff
@@ -1125,7 +1143,19 @@ p5.prototype.model = function(model) {
       this._renderer.createBuffers(model.gid, model);
     }
 
-    this._renderer.drawBuffers(model.gid);
+    // If model has textures for each texture update index buffer and invoke draw call.
+    if(model.hasTextures) {
+      this._renderer.updateIndexBuffer(model.gid, model.faces);
+      this._renderer.drawBuffers(model.gid);
+      for (let material of Object.keys(model.textures)) {
+        const texture = model.textures[material];
+        this._renderer.updateIndexBuffer(model.gid, texture.faces);
```

Comment:
Is there a way for us to do this by having multiple index buffers that we swap between? [In the past we've found that on some computers, the cpu-to-gpu data transfer is by far the biggest bottleneck](https://github.com/processing/p5.js/pull/6162), so we've tried to minimize the data sent back and forth. Currently, `p5.Geometry` is the way to cache this reused data to avoid any transfer after the first time you draw it, so ideally we'd keep that property.